map_test.go

```diff
@@ -2,128 +2,513 @@ package swiss
 
 import (
 	"fmt"
-	"math/bits"
 	"math/rand"
-	"strconv"
+	"sort"
 	"testing"
+	"unsafe"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-// TODO(peter)
-// - Insertion
-// - Deletion
-// - Lookup hit vs lookup miss
-// - Resizing
-// - Dropping tombstones without resizing
-// - Iteration
-// - Iteration while mutating the map
+// toBuiltinMap returns the elements as a map[K]V. Useful for testing.
+func (m *Map[K, V]) toBuiltinMap() map[K]V {
+	r := make(map[K]V)
+	m.All(func(k K, v V) bool {
+		r[k] = v
+		return true
+	})
+	return r
+}
 
-func TestBasic(t *testing.T) {
-	m := New[int, int](0)
-	for i := 0; i < 10; i++ {
-		m.Put(i, i+10)
-		if v, ok := m.Get(i); ok {
-			fmt.Printf("%d\n", v)
+// TODO(peter): Extracting a random element might be generally useful. Should
+// this be promoted to the public API?
+func (m *Map[K, V]) randElement() (key K, value V, ok bool) {
```

Comment:
This should probably take a `rand.Rand`, especially if we consider making it a public API. It would also help to make tests reproducible if we print out the seed.

Reply:
Agreed about passing in a `rand.Rand`. I'll probably do that regardless in a follow-on PR. I have too many stacked on top of each other right now.